File: backend/app/schemas/history.py
from datetime import datetime

from pydantic import BaseModel, Field

from app.schemas.qa import QaEvidence


class HistoryItem(BaseModel):
    """问答历史记录。"""

    id: int
    question: str
    answer: str
    mode: str
    confidence: float
    elapsed_ms: int
    graph_path_count: int
    text_evidence_count: int
    graph_paths: list[list[str]] = Field(default_factory=list)
    text_evidence: list[QaEvidence] = Field(default_factory=list)
    created_at: datetime


class HistoryListResponse(BaseModel):
    """问答历史分页响应。"""

    items: list[HistoryItem]
    total: int
    page: int
    page_size: int


class HistoryStats(BaseModel):
    """问答历史统计。"""

    total: int
    avg_confidence: float | None = None
    avg_elapsed_ms: float | None = None
    kg_rag_count: int = 0


class HistoryQuery(BaseModel):
    """历史查询参数约束。"""

    page: int = Field(default=1, ge=1)
    page_size: int = Field(default=10, ge=1, le=100)
    keyword: str = ""

File: frontend/src/api/history.ts
import { apiClient } from "./client";
import type { ApiResponse, QaEvidence } from "./types";

export interface HistoryItem {
  id: number;
  question: string;
  answer: string;
  mode: string;
  confidence: number;
  elapsed_ms: number;
  graph_path_count: number;
  text_evidence_count: number;
  graph_paths: string[][];
  text_evidence: QaEvidence[];
  created_at: string;
}

export interface HistoryListResponse {
  items: HistoryItem[];
  total: number;
  page: number;
  page_size: number;
}

export interface HistoryStats {
  total: number;
  avg_confidence: number | null;
  avg_elapsed_ms: number | null;
  kg_rag_count: number;
}

export async function fetchHistory(page = 1, pageSize = 10, keyword = ""): Promise<HistoryListResponse> {
  const response = await apiClient.get<ApiResponse<HistoryListResponse>>("/api/history", {
    params: { page, page_size: pageSize, keyword },
  });
  return response.data.data;
}

export async function fetchHistoryStats(): Promise<HistoryStats> {
  const response = await apiClient.get<ApiResponse<HistoryStats>>("/api/history/stats");
  return response.data.data;
}

export async function deleteHistoryItem(id: number): Promise<void> {
  await apiClient.delete(`/api/history/${id}`);
}
